parquet-column/src/main/java/org/apache/parquet/schema/Types.java

```diff
@@ -566,6 +571,13 @@ private Optional<Boolean> checkBinaryPrimitiveType(LogicalTypeAnnotation logical
             return Optional.of(true);
           }
 
+          private Optional<Boolean> checkFloat16BinaryPrimitiveType(int l, LogicalTypeAnnotation logicalTypeAnnotation) {
+            Preconditions.checkState(
+              primitiveType == PrimitiveTypeName.BINARY && length == l,
```

Comment:
Float16 annotates FIXED_LENGTH_BYTE_ARRAY type not BINARY. This method should be removed.